examples/threading/threading.c

```diff
@@ -13,8 +13,16 @@ void* threadfunc(void* thread_param)
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
-    return thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+    printf("%s", thread_func_args->message);
+    usleep(thread_func_args->wait_to_obtain_ms*1000);
+    pthread_mutex_lock (thread_func_args->mutex);
+    usleep(thread_func_args->wait_to_release_ms*1000);
+    pthread_mutex_unlock (thread_func_args->mutex);
+    
+    thread_func_args->thread_complete_success = true;
```

Comment:
@Jahnavi-pinnamaneni what will happen if you return success = true in case of mutex lock/unlock or usleep failure?